src/reactors/PriorityOrderReactorQuoter.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {BaseReactor} from "./BaseReactor.sol";
+import {Permit2Lib} from "../lib/Permit2Lib.sol";
+import {ResolvedOrderLib} from "../lib/ResolvedOrderLib.sol";
+import {PriorityOrderLib, PriorityOrder, PriorityInput, PriorityOutput} from "../lib/PriorityOrderLib.sol";
+import {PriorityFeeLib} from "../lib/PriorityFeeLib.sol";
+import {CosignerLib} from "../lib/CosignerLib.sol";
+import {SignedOrder, ResolvedOrder} from "../base/ReactorStructs.sol";
+import {IPermit2} from "permit2/src/interfaces/IPermit2.sol";
+
+/// @notice Reactor for priority orders
+/// @dev only supported on chains which use priority fee transaction ordering
+contract PriorityOrderReactorQuoter is BaseReactor {
+    using Permit2Lib for ResolvedOrder;
+    using ResolvedOrderLib for ResolvedOrder;
+    using PriorityOrderLib for PriorityOrder;
+    using PriorityFeeLib for PriorityInput;
+    using PriorityFeeLib for PriorityOutput[];
+
+    /// @notice thrown when an order's deadline is in the past
+    error InvalidDeadline();
+    /// @notice thrown when an order's auctionStartBlock is in the future
+    error OrderNotFillable();
+    /// @notice thrown when an order's nonce has already been used
+    error OrderAlreadyFilled();
+    /// @notice thrown when an order's input and outputs both scale with priority fee
+    error InputOutputScaling();
+    /// @notice thrown when tx gasprice is less than block.basefee
+    error InvalidGasPrice();
+
+    mapping(bytes32 => uint256) public orderAuctionStartBlocks;
+
+    constructor(IPermit2 _permit2, address _protocolFeeOwner) BaseReactor(_permit2, _protocolFeeOwner) {}
+
+    /// @inheritdoc BaseReactor
+    function _resolve(SignedOrder calldata signedOrder)
```

Comment:
probably need a public quote function? Can have a quoter interface that both this and the exiting quoter implements